proposals/p1964.md

```diff
@@ -0,0 +1,372 @@
+# Character literals
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1964)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+    -   [Types](#types)
+    -   [Operations](#operations)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [No distinct character types](#no-distinct-character-types)
+    -   [No distinct character literal](#no-distinct-character-literal)
+    -   [Supporting prefix declarations](#supporting-prefix-declarations)
+    -   [Allowing numeric escape sequences](#allowing-numeric-escape-sequences)
+    -   [Supporting formulations of grapheme clusters and non-code-point code-units](#supporting-formulations-of-grapheme-clusters-and-non-code-point-code-units)
+-   [Future Work](#future-work)
+    -   [UTF code unit types proposal](#utf-code-unit-types-proposal)
+
+<!-- tocstop -->
+
+## Abstract
+
+This proposal specifies lexical rules for constant characters in Carbon:
+
+Put character literals in single quotes, like `'a'`. Character literals work
+like numeric literals:
+
+-   Every different literal value has its own type.
+-   The bit width is determined by the type of the variable the literal is
+    assigned to, not the literal itself.
```

Comment:
The wording of this bullet point was a little hard for me to follow. What do folks think about something more like the following?

```suggestion
-   The literal itself doesn't have a bit width as a consequence. Instead, variables use explicitly sized character types and character literals can be converted to these types when representable.
```